channeldb/kvdb/etcd/stm.go

```diff
@@ -235,22 +235,32 @@ func runSTM(s *stm, apply func(STM) error) error {
 		err     error
 	)
 
-	// In a loop try to apply and commit and roll back
-	// if the database has changed (CommitError).
+Loop:
+	// In a loop try to apply and commit and roll back if the database has
+	// changed (CommitError).
 	for {
-		// Abort STM if there was an application error.
+		select {
+		// Check if the STM is aborted and break the retry loop if it is.
+		case <-s.options.ctx.Done():
+			err = fmt.Errorf("aborted")
+			break Loop
+
+		default:
+		}
+
+		// Apply the transaction closure and abort the STM if there was an
+		// application error.
 		if err = apply(s); err != nil {
-			break
+			break Loop
```

Comment:
wouldn't this already break the loop without the label? maybe it's just done for consistency?

Reply:
Just for consistency. Can remove if you think it's noise.

Reply:
i think it's fine, i think it makes it clear that they all exit the same way